client/softlayer_client.go

```diff
@@ -22,33 +22,34 @@ const (
 	SL_GO_NON_VERBOSE  = "SL_GO_NON_VERBOSE"
 )
 
-type softLayerClient struct {
+type SoftLayerClient struct {
 	username string
 	apiKey   string
 
 	templatePath string
 
-	httpClient *http.Client
+	// HTTPClient is used for HTTP requests. By default DefaultHTTPClient is
+	// being used.
+	HTTPClient *http.Client
 
 	softLayerServices map[string]softlayer.Service
 
 	nonVerbose bool
 }
 
-func NewSoftLayerClient(username, apiKey string) *softLayerClient {
-	pwd, _ := os.Getwd()
-	slc := &softLayerClient{
+func NewSoftLayerClient(username, apiKey string) *SoftLayerClient {
+	pwd, err := os.Getwd()
+	if err != nil {
+		panic(err) // this should be handled by the user
```

Comment:
I would do the same and I agree with you, but I would do this with another PR. Because this requires us to change and rethink all the tests and the current usages of this constructor. Also I think it's better in long term to improve the state of this package.